src/logic/hooks/useEstimateTransactionGas.tsx

```diff
@@ -115,7 +112,18 @@ export const useEstimateTransactionGas = ({
   const { address: safeAddress = '', threshold = 1, currentVersion: safeVersion = '' } = useSelector(currentSafe) ?? {}
   const { account: from, smartContractWallet, name: providerName } = useSelector(providerSelector)
 
-  const canTxExecute = useCanTxExecute(isExecution, manualSafeNonce, preApprovingOwner, txConfirmations?.size)
+  const recommendedNonce = useGetRecommendedNonce(safeAddress)
+  const { nonce: currentSafeNonce } = useSelector(currentSafe)
+
+  const canTxExecute = calculateCanTxExecute(
+    currentSafeNonce,
+    preApprovingOwner ?? '',
```

Comment:
Are these two definitely `undefined` or `null`?

Reply:
I reverted the change since we have fallbacks for both when calling `useCanTxExecute`. In some places where we call `useCanTxExecute` they are undefined.